src/lib/oauth2-server.ts

```diff
@@ -15,24 +15,26 @@
 
 /**
  * OAuth2 HTTP Server library
+ *
  * @module lib/oauth2-server
  */
 
-'use strict';
-
-const { URL } = require('url');
-const net = require('net');
-const HttpServer = require('./http-server');
-const OAuth2Issuer = require('./oauth2-issuer');
-const OAuth2Service = require('./oauth2-service');
+import { URL } from 'url';
+import { isIP, AddressInfo } from 'net';
+import { Server } from 'http';
 
-const issuer = Symbol('issuer');
-const service = Symbol('service');
+import { HttpServer } from './http-server';
+import { OAuth2Issuer } from './oauth2-issuer';
+import { OAuth2Service } from './oauth2-service';
+import { AssertionError } from 'assert';
 
 /**
  * Represents an OAuth2 HTTP server.
  */
-class OAuth2Server extends HttpServer {
+export class OAuth2Server extends HttpServer {
+  private _service: OAuth2Service;
```

Comment:
Can't use ECMAScript private fields because of `A 'super' call must be the first statement in the constructor when a class contains initialized properties, parameter properties, or private identifiers.` ts error message :-(